File: todolist_client/lib/models/task.dart
class Task {
  String creatorId;
  DateTime date;
  String title;
  bool done;
  String id;

  Task({this.title, this.date, this.creatorId, this.done, this.id});

  factory Task.fromMap(Map<String, dynamic> map) {
    return Task(
        title: map["title"],
        date: (map["date"] != null)
            ? DateTime.fromMillisecondsSinceEpoch(map["date"])
            : null,
        done: map["done"],
        creatorId: map["creatorId"],
        id: map["id"]);
  }

  Map<String, dynamic> toMap() {
    return {
      "title": title,
      "date": date?.millisecondsSinceEpoch,
      "done": done,
      "creatorId": creatorId,
      "id": id
    };
  }
}

File: todolist_server/models/task.js
const mongoose = require("mongoose");

const taskSchema = mongoose.Schema({
  creatorId: { type: String, required: true },
  date: { type: Number, required: true },
  title: { type: String, required: true },
  done: { type: Boolean, required: true }
});

module.exports = mongoose.model("Task", taskSchema);
